src/lib/search-params.ts

```diff
@@ -0,0 +1,137 @@
+import {
+  SearchOptions,
+  SortBy as SortBySearchOption,
+  SortOrder,
+} from '@/lib/dataset-fetcher';
+
+export enum SortBy {
+  RelevanceDesc = 'relevanceDesc',
+  NameAsc = 'nameAsc',
+  NameDesc = 'nameDesc',
+}
+
+export const defaultSortBy = SortBy.RelevanceDesc;
+
+const sortMapping = {
+  [SortBy.RelevanceDesc]: {
+    sortBy: SortBySearchOption.Relevance,
+    sortOrder: SortOrder.Descending,
+  },
+  [SortBy.NameAsc]: {
+    sortBy: SortBySearchOption.Name,
+    sortOrder: SortOrder.Ascending,
+  },
+  [SortBy.NameDesc]: {
+    sortBy: SortBySearchOption.Name,
+    sortOrder: SortOrder.Descending,
+  },
+};
+
+interface ToSearchParamsProps {
+  query?: SearchOptions['query'];
+  offset?: SearchOptions['offset'];
+  filters: {
+    licenses?: string[];
+    publishers?: string[];
+  };
+  sortBy?: SortBy;
+}
+
+export function getUrlWithSearchParams({
+  query,
+  offset,
+  filters: {licenses, publishers},
+  sortBy,
+}: ToSearchParamsProps) {
+  const searchParams: {[key: string]: string} = {};
+
+  if (query) {
+    searchParams.query = query;
+  }
+
+  if (licenses?.length) {
+    searchParams.licenses = licenses.join(',');
+  }
+
+  if (publishers?.length) {
+    searchParams.publishers = publishers.join(',');
+  }
+
+  if (offset) {
+    searchParams.offset = `${offset}`;
+  }
+
+  if (sortBy && sortBy !== defaultSortBy) {
+    searchParams.sortBy = sortBy;
+  }
+
+  const encodedSearchParams = new URLSearchParams(searchParams).toString();
+
+  if (encodedSearchParams) {
+    return '/?' + encodedSearchParams;
+  } else {
+    return '/' + encodedSearchParams;
+  }
+}
+
+interface FromSearchParamsToSearchOptionsProps {
+  publishers?: string;
+  licenses?: string;
+  query?: string;
+  offset?: string;
+  sortBy?: SortBy;
+}
+
+interface Response {
+  searchOptions: SearchOptions;
+  sortBy: SortBy;
```

Comment:
Why is `sortBy` a separate property (and not a part of the search options)? (Perhaps add a comment?)

Reply:
This is the sortBy used by the client. This is a combination of sortBy and sortOrder in the searchOptions. Let me see how I can make this more clear

Reply:
I decided to remove this return value and only return searchOptions. This just makes the whole function confusing. The client can figure out later what value it needs. That this function can just do one thing.